text/0000-sync-networked-vars.md

```diff
@@ -0,0 +1,110 @@
+#### Local tick
+The tick number at which a variable was modified, relative to the tick system on this machine. If I get an update for a variable I changed, this tell me: when did I change it?
+
+#### Remote tick
+The tick number at which a variable was relayed, relative to the tick system on the remote machine. If I get an update from the server, this tells me the local tick on the server when it sent the update.
+
+#### Network Tick wrapping 
+
+The tick value sent will be a rolling value, probably 8- or 16-bit. This will be hidden internally, but will keep the bandwidth usage in check.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+We suggest changing:
+
+- in `Runtime/NetworkedVar/INetworkedVar.cs`, `INetworkedVar` `ReadField()` to become
+
+      public void ReadField(Stream stream, ushort localTick, ushort remoteTick);
+
+- in `Runtime/NetworkedVar/INetworkedVar.cs`, `INetworkedVar` `ReadDelta()` to become
+
+      public void ReadDelta(Stream stream, bool keepDirtyDelta, ushort localTick, ushort remoteTick);
+
+- `NetworkedVar` to include two new properties:
+
+      public ushort LocalTick { get; internal set; }
+      public ushort RemoteTick { get; internal set; }
+      
+ - `SendTickrate` and `LastSyncedTime` to be removed from `NetworkedVar`
+
+Below diagram shows a couple NetworkedVariable updates with example tick numbers.
+
+![](0000-sync-networked-vars/netvars.png)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This increases slightly the bandwidth usage of _individual_ NetworkedVariable update it also renders the NetworkedVariables system dependent on the tick system.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This proposal makes the NetworkedVariables a bit more complex. But it sets a foundation over which other features can be built without causing the NetworkedVariables to be modified further. Snapshotting, client-side interpolation, lag compensation, replay and rollback are examples features that will leverage this. 
+
+# Prior art
+[prior-art]: #prior-art
+
+This is a pretty typical setup seen in other proprietary network engines.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+Do we require the tick to advance at the same rate on the client and server machine. Can we support an asymmetry in tick rate between client and server? Is this important?
+
+Different clients will have a different relative ticks. When the host and client receive the NetworkedVariables, they can flip the local and remote tick. As such, if Alice sends an update to the Host and gets her value back, the local tick is valid for her. This update would not be valid for a different client, though. As such, when the snapshot and interpolation systems are fleshed out, a fix might be needed. It will probably involve marking the NetworkedVariables with a ClientId or client index. We prefer to leave this detail out of this RFC, for now. 
```

Comment:
As long as we know who owns those tick numbers, we can still workaround it. I'm taking control over an object, it's perfectly fine to throw the latest values away and start relying on my own ticks, values etc. Maybe a smooth interpolation/transition between the latest not-in-control state to the current now-in-control state be needed but that wouldn't dramatically affect the implementation IMHO.